File: python/src/problems1_100/length_of_last_word.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def length_of_last_word(self, s: str) -> bool:
        pass


class Default(Problem):
    @override
    def length_of_last_word(self, s: str) -> bool:
        chars: list[str] = list(s)
        for i in range(len(chars) // 2):
            temp = chars[i]
            chars[i] = chars[len(chars) - 1 - i]
            chars[len(chars) - 1 - i] = temp

        count = 0
        start_counting = False
        for c in chars:
            if c == ' ':
                if start_counting:
                    break
            else:
                start_counting = True
            if start_counting:
                count += 1

        return count


SOLUTIONS: list[Problem] = [Default()]

File: typescript/src/problems1_100/length-of-last-word.ts
class Problem {
  lengthOfLastWord(s: string): number {
    throw new Error(`Not implemented (${s}).`);
  }
}

class Default extends Problem {
  lengthOfLastWord(s: string): number {
    const chars: string[] = s.split('');
    for (let i = 0; i < chars.length / 2; i++) {
      const temp: string = chars[i];
      chars[i] = chars[chars.length - 1 - i];
      chars[chars.length - 1 - i] = temp;
    }

    let count: number = 0;
    let startCounting: boolean = false;
    for (const c of chars) {
      if (c === ' ') {
        if (startCounting) {
          break;
        }
      } else {
        startCounting = true;
      }
      if (startCounting) {
        count++;
      }
    }
    return count;
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;
